### ISSUE TITLE
Handle KEYCODE_DPAD_CENTER and KEYCODE_ENTER

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#1670]
http://developer.android.com/guide/topics/ui/accessibility/apps.html:
>Handling directional controller clicks
>
>On most devices, clicking a view using a directional controller sends a KeyEvent with KEYCODE_DPAD_CENTER to the view currently in focus. All standard Android views already handle KEYCODE_DPAD_CENTER appropriately. When building a custom View control, make sure this event has the same effect as touching the view on the touchscreen.
>
>Your custom control should also treat the KEYCODE_ENTER event the same as KEYCODE_DPAD_CENTER. This approach makes interaction from a full keyboard much easier for users.

**Comment by goderbauer** [comment]
New location of quote: https://developer.android.com/guide/topics/ui/accessibility/custom-views.html

**Comment by gspencergoog** [comment]
Yes, along with lots of other keycode support.

**Comment by GaryQian** [comment]
@gspencergoog Does your keypad support work resolve this issue?

**Comment by gspencergoog** [comment]
No, I don't think so, because I don't think those keys are sent to the framework from the engine yet.

**Comment by mossmana** [comment]
Official docs have moved to: https://developer.android.com/guide/topics/ui/accessibility/custom-views

@gspencergoog I was considering picking this issue up, but would it make sense to wait until https://github.com/flutter/flutter/issues/136419 is complete?

**Comment by gspencergoog** [comment]
Well... it's up to you.  That won't be "finished" for more than a year, since once it's deprecated, we need to wait a year to remove the redundant code. So this bug will be open another year before being fixed.  The down side of fixing it now is that you'll probably have to fix it for both kinds of events. It's not likely to be too complicated, however.

Personally, I'd just fix it now regardless and we'll remove the redundant code when we remove the rest of it in a year.

**Comment by mossmana** [comment]
I'll take a preliminary look at this issue. Tracking findings in https://docs.google.com/document/d/1Y984br0oLq2rkyQku4ve_zpR8lOeETb24ulCJN6yUls/edit?usp=sharing
